From 0da5a58823ec3d3a710551d893a1414a86ffc940 Mon Sep 17 00:00:00 2001
From: Ken Ma <make@marvell.com>
Date: Thu, 16 Mar 2017 10:27:01 +0800
Subject: [PATCH 0884/1345] xhci-plat: introduce a new property for xhci reset
 after resume

commit  c92cd23c65f4b2de332a0051756ceb5b8fa29aac from
https://github.com/MarvellEmbeddedProcessors/linux-marvell.git

- Some SOCs such as Armada 3700 must force XHCI reset after resume;
- This patch introduces a new fdt property "needs-reset-on-resume"
  which is set to force XHCI reset after resume;
- When this property is set, xhci quirks will set XHCI_RESET_ON_RESUME
  bit, and then XHCI resume will do reset.

Change-Id: I38ca3763f54fbb721e408e1882923c4a84b76856
Signed-off-by: Ken Ma <make@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/37493
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
Reviewed-by: Hanna Hawa <hannah@marvell.com>
Reviewed-by: Victor Gu <xigu@marvell.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 Documentation/devicetree/bindings/usb/usb-xhci.txt |    3 +++
 drivers/usb/host/xhci-plat.c                       |    3 +++
 2 files changed, 6 insertions(+)

diff --git a/Documentation/devicetree/bindings/usb/usb-xhci.txt b/Documentation/devicetree/bindings/usb/usb-xhci.txt
index 03f0afd..38ba6f3 100644
--- a/Documentation/devicetree/bindings/usb/usb-xhci.txt
+++ b/Documentation/devicetree/bindings/usb/usb-xhci.txt
@@ -36,6 +36,8 @@ Optional properties:
   - phys: phandle + phy specifier pair, reference to the USB PHYs
   - phy-names: can be "usb2", "usb3" when "separated-phys-for-usb2-usb3" is
     set or "usb" when "separated-phys-for-usb2-usb3" is not set.
+  - needs-reset-on-resume: boolean, set this to force XHCI reset after resume,
+    for example, Armada 3700 must force XHCI reset after resume
 
 Examples:
 	usb@f0931000 {
@@ -55,4 +57,5 @@ Examples:
 		       <&a3700_comphy 1 COMPHY_USB3>; /* usb3 comphy */
 		phy-names = "usb2", "usb3";
 		separated-phys-for-usb2-usb3;
+		needs-reset-on-resume;
 	};
\ No newline at end of file
diff --git a/drivers/usb/host/xhci-plat.c b/drivers/usb/host/xhci-plat.c
index 932dd3b..5025359 100644
--- a/drivers/usb/host/xhci-plat.c
+++ b/drivers/usb/host/xhci-plat.c
@@ -74,6 +74,9 @@ static void xhci_plat_quirks(struct device *dev, struct xhci_hcd *xhci)
 	 * dev struct in order to setup MSI
 	 */
 	xhci->quirks |= XHCI_PLAT;
+
+	if (of_property_read_bool(dev->of_node, "needs-reset-on-resume"))
+		xhci->quirks |= XHCI_RESET_ON_RESUME;
 }
 
 /* called during probe() after chip reset completes */
-- 
1.7.9.5

